packages/components/src/navigator/navigator-screen/component.tsx

```diff
@@ -65,12 +72,40 @@ function NavigatorScreen( props: Props, forwardedRef: Ref< any > ) {
 		[ className ]
 	);
 
-	// This flag is used to only apply the focus on mount when the actual path changes.
-	// It avoids the focus to happen on the first render.
-	const [ hasPathChanged, setHasPathChanged ] = useState( false );
+	// Focus restoration
+	const [ isFirstRender, setIsFirstRender ] = useState( true );
 	useEffect( () => {
-		setHasPathChanged( true );
-	}, [ path ] );
+		// Only attempt to restore focus:
+		// - after the first render (to avoid moving focus on page load)
+		// - when the screen becomes visible
+		// - the wrapper ref has been assigned
+		if ( isFirstRender || ! isMatch || ! wrapperRef.current ) {
+			setIsFirstRender( false );
+			return;
+		}
+
+		let elementToFocus: HTMLElement | null = null;
+
+		// When navigating back, if a selector is provided, use it to look for the
+		// target element (assumed to be a node inside the current NavigatorScreen)
+		if ( location.isBack && location.focusRestorationSelector ) {
```

Comment:
`location.focusRestorationSelector` is a weird name, should we name it more "target" or "focusTarget" something like that?

Reply:
100% agree that `focusRestorationSelector` is not a good variable name — I gave up coming up with a good name since I wasn't sure that this "selector-based" approach was going to stick.

I thought about `focusTarget`, but in case we go with the current approach, should we make it a bit more obvious that the variable needs to represent a CSS selector? Maybe `focusTargetSelector` ?

Reply:
I [renamed it](https://github.com/WordPress/gutenberg/pull/37223/commits/59ab94e36979e04f8973864512eb91744581f9df) to `focusTargetSelector` for now — happy to change this variable's to something else in case we change approach and stop using a CSS selector